src/coreclr/tools/Common/Internal/Runtime/ReadyToRunConstants.cs

```diff
@@ -165,6 +165,8 @@ public enum ReadyToRunFixupKind
         Check_IL_Body              = 0x35, /* Check to see if an IL method is defined the same at runtime as at compile time. A failed match will cause code not to be used. */
         Verify_IL_Body             = 0x36, /* Verify an IL body is defined the same at compile time and runtime. A failed match will cause a hard runtime failure. */
 
+        RvaFieldAddress            = 0x37, /* Used in Crossgen2 to address RVA fields in their original MSIL images */
```

Comment:
Can we just use existing `FieldAddress` fixup for this?

Reply:
I hoped we could but it looks like <code>FieldAddress</code> is specific to instance fields and cannot be used as a precode. The newly introduced fixup is the most similar to <code>StaticFieldAddress</code> but that one does something weird around storing the result next to the fixup cell, I guess that's something specific to runtime JIT behavior.

Reply:
Please see

https://github.com/dotnet/runtime/blob/f924d6b5f69792f33a507b47a1fc870a185ad638/src/coreclr/vm/jitinterface.cpp#L13825

Reply:
I do not see the problem. The comment on `FieldAddress` says that it is for static fields:

FieldAddress = 0x20, // For accessing a cross-module static fields

If FieldAddress cannot be used as precode fixup, we can make it usable as precode fixup. We have other fixups like that. The encoding is designed to allow the same `ReadyToRunFixupKind` to be used in different modes.

Reply:
OK, thanks for the feedback, I'll take look into making <code>FieldAddress</code> work for this purpose.

Reply:
> Please see
> https://github.com/dotnet/runtime/blob/f924d6b5f69792f33a507b47a1fc870a185ad638/src/coreclr/vm/jitinterface.cpp#L13825

You can delete `ENCODE_STATIC_FIELD_ADDRESS` while you are on it. It looks like a left-over from fragile NGen.

Reply:
Fixed in 2nd commit, thanks for the suggestions!